Backend/src/ml_lib/sub_models/qaqc_main_model.py

```diff
@@ -71,32 +71,8 @@ def first_run(self, df: pd.DataFrame, dim: str | None) -> pd.DataFrame:
         id_df = pd.DataFrame(data=d)
```

Comment:
I know that this is not added to this commit, but I think you could clean up the concatinations by writing this:

```python
ids = [self.get_id(question) for question in questions]
df["id"] = pd.Series(ids)
```

If you add this you don't need the variable `d` and you can also remove the line `combined_df = pd.concat([combined_df, id_df], axis=1)`.

I think this is fitting, given that your code cleans up this part, even though I should have caught this in a previous review.

Reply:
great catch, will do in my current branch for issue 69